plugins/out_s3/s3_parquet.c

```diff
@@ -0,0 +1,500 @@
+                                      json_buffer, json_size,
+                                      (void **)&parquet_buffer, &parquet_size);
+
+    if (ret < 0) {
+        flb_plg_error(ctx->ins,
+                     "parquet conversion failed for batch '%s'",
+                     batch->tag);
+        flb_free(json_buffer);
+
+        /* Return error to trigger Fluent Bit retry mechanism */
+        return -1;
+    }
+
+    flb_free(json_buffer);
+
+    /* Record end time and calculate elapsed time */
+    flb_time_get(&end_time);
+    {
+        struct flb_time diff_time;
+        flb_time_diff(&end_time, &start_time, &diff_time);
+        elapsed_ms = flb_time_to_nanosec(&diff_time) / 1000000;
+    }
+
+    flb_plg_debug(ctx->ins,
+                 "parquet conversion: %zu bytes -> %zu bytes, %"PRIu64"ms",
+                 json_size, parquet_size, elapsed_ms);
+
+    /* 2.5. Apply additional compression if configured (gzip/zstd) */
+    if (ctx->compression == FLB_AWS_COMPRESS_GZIP ||
+        ctx->compression == FLB_AWS_COMPRESS_ZSTD) {
+        void *compressed_buffer = NULL;
+        size_t compressed_size = 0;
+        size_t pre_compress_size = parquet_size;
+
+        ret = flb_aws_compression_compress(ctx->compression,
+                                          parquet_buffer, parquet_size,
+                                          &compressed_buffer, &compressed_size);
+        if (ret < 0) {
+            flb_plg_error(ctx->ins,
+                         "failed to apply %s compression for batch '%s'",
+                         ctx->compression == FLB_AWS_COMPRESS_GZIP ? "gzip" : "zstd",
+                         batch->tag);
+            flb_free(parquet_buffer);
+            return -1;
+        }
+
+        flb_free(parquet_buffer);
+        parquet_buffer = compressed_buffer;
+        parquet_size = compressed_size;
+
+        flb_plg_debug(ctx->ins,
+                     "%s compression: %zu bytes -> %zu bytes",
+                     ctx->compression == FLB_AWS_COMPRESS_GZIP ? "gzip" : "zstd",
+                     pre_compress_size, compressed_size);
+    }
+
+    /* 3. Upload Parquet data */
+    if (ctx->use_put_object == FLB_TRUE) {
+        /* Check PutObject size limit */
+        if (parquet_size > MAX_FILE_SIZE_PUT_OBJECT) {
+            flb_plg_error(ctx->ins,
+                         "parquet size %zu exceeds 1GB limit for use_put_object mode",
+                         parquet_size);
+            flb_free(parquet_buffer);
+            return -1;
+        }
+
+        /* Use PutObject */
+        ret = s3_put_object(ctx, batch->tag, batch->create_time,
+                          parquet_buffer, parquet_size);
+        if (ret < 0) {
+            flb_plg_error(ctx->ins,
+                         "failed to upload batch '%s'",
+                         batch->tag);
+            flb_free(parquet_buffer);
+            return -1;
+        }
+    } else if (parquet_size > MAX_FILE_SIZE_PUT_OBJECT) {
```

Comment:
We need to use a newline like as:

```suggestion
    } 
    else if (parquet_size > MAX_FILE_SIZE_PUT_OBJECT) {
```